### ISSUE TITLE
Variables Persist Between Cells section is confuse

### ISSUE THREAD

**Issue by rgaiacs** [swcarpentry/python-novice-gapminder#184]
"Variables Persist Between Cells" section says

> Variables defined in one cell exist in all other cells once executed, so the relative location of cells in the notebook do not matter (i.e., cells lower down can still affect those above). Remember: Notebook cells are just a way to organize a program: as far as Python is concerned, all of the source code is one long set of instructions.

I think the text very confuse but I don't know how to make it clear without adding technical details.

**Comment by drmarkwslater** [comment]
How about this as a replacement? Maybe a bit wordy but I've found this to be quite an important gotcha for new users of Jupyter notebooks

Be aware that it is the order of **execution** of cells that is important in a Jupyter notebook, not the order in which they appear. Python will remember **all** the code that was run previously, including any variables you have defined, irrespective of the order in the notebook. Therefore if you define variables lower down the notebook and then (re)run cells further up, those defined further down will still be present. As an example, create 2 cells with the following content, in this order:
```
print(myval)
```
```
myval = 1
```
If you execute this in order, the first cell will give an error. However, if you run the first cell **after** the second cell it will print out ‘1’. To prevent confusion, it can be helpful to use the `Kernel` -> `Restart & Run All` option which clears the interpreter and runs everything from a clean slater going top to bottom.

**Comment by vahtras** [comment]
I like it

**Comment by drmarkwslater** [comment]
Great 😄 Shall I prepare a PR for it?

**Comment by vahtras** [comment]
Please do!
